cometd-java/cometd-java-common/src/main/java/org/cometd/common/Z85.java

```diff
@@ -60,18 +60,18 @@ public String encodeBytes(byte[] bytes) {
             int padding = 4 - (remainder == 0 ? 4 : remainder);
             StringBuilder result = new StringBuilder();
             long value = 0;
+            char[] inv = new char[5];
             for (int i = 0; i < length + padding; ++i) {
                 boolean isPadding = i >= length;
-                value = value * 256 + (isPadding ? 0 : bytes[i] & 0xFF);
+                value = (value << 8) + (isPadding ? 0 : bytes[i] & 0xFF);
                 if ((i + 1) % 4 == 0) {
```

Comment:
Since you're doing this, `(i + 1) % 4 == 0` can be replaced with (i + 1) & 3 == 0`?

Reply:
Mhh… the first is "make space for one byte" the other is "do this every 4 bytes"… and is very similar to `(i + 1) % 5 == 0` a few lines later in the decoder, so personally I wouldn't do it. Speed-wise I think there is no difference, the compiled should be smart enough to see the equivalence.

Reply:
I stand corrected: I checked and _it does_ do a little bit of change in speed; so, yes, will do it.